src/storage/src/source/kafka.rs

```diff
@@ -921,16 +932,19 @@ impl SourceRender for KafkaSourceConnection {
                         (health_status.kafka.take(), health_status.ssh.take())
                     };
                     if let Some(status) = kafka_status {
-                        health_output.give(
-                            &health_cap,
-                            HealthStatusMessage {
-                                index: 0,
-                                namespace: Self::STATUS_NAMESPACE.clone(),
-                                update: status,
-                            },
-                        );
+                        for (output, status) in outputs.iter().repeat_clone(status) {
+                            health_output.give(
+                                &health_cap,
+                                HealthStatusMessage {
+                                    index: output.output_index,
+                                    namespace: Self::STATUS_NAMESPACE.clone(),
+                                    update: status,
+                                },
+                            );
+                        }
                     }
                     if let Some(status) = ssh_status {
+                        // We only report ssh status errors to output 0 (the primary source).
```

Comment:
I would also add this to all outputs

Reply:
sounds good, will update everywhere we report status just to `index: 0` and output to all outputs instead